cpp/src/io/parquet/page_enc.cu

```diff
@@ -1822,6 +1848,7 @@ dremel_data get_dremel_data(column_view h_col,
                             // TODO(cp): use device_span once it is converted to a single hd_vec
                             rmm::device_uvector<uint8_t> const& d_nullability,
                             std::vector<uint8_t> const& nullability,
+                            bool output_as_byte_array,
```

Comment:
I know this isn't part of the review, but the example in the documentation for `get_dremel_data()` seems incorrect at line#1738 (now #1764):
```c++
 * Given a LIST column of type `List<List<int>>` like so:
 * ```
 * col = {
 *    [],
 *    [[], [1, 2, 3], [4, 5]],
 *    [[]]
 * }
 * ```
 * We can represent it in cudf format with two level of offsets like this:
 * ```
 * Level 0 offsets = {0, 0, 3, 5, 6}
 * Level 1 offsets = {0, 0, 3, 5, 5}
 * Values          = {1, 2, 3, 4, 5}
 * ```
```
The `Level 0` offset values can't exceed `4`, since `Level 1` has only 4 ranges (i.e. 5 offsets).
I'll try make sense of this at a later date, but I'm not sure I could follow along.

Reply:
https://github.com/rapidsai/cudf/issues/11396 :)